product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -131,6 +131,7 @@ def _select_customerinfo(
         params.update({"partner_id": partner.id})
         domain = self._prepare_domain_customerinfo(params)
         res = self.env["product.customerinfo"].search(
-            domain, order="product_id, sequence", limit=1
-        )
-        return res
+            domain
+        ).sorted(lambda s: (s.sequence, s.min_qty, s.price, s.id))
```

Comment:
@nalejhandro Why changing the order from previous version ?

EDIT: Ok, saw the commit message. Could you maybe put that change in a separate commit ?